src/parser.cpp

```diff
@@ -95,20 +96,20 @@ namespace ada::parser {
   /**
    * @see https://url.spec.whatwg.org/#concept-opaque-host-parser
    */
-  std::optional<ada::url_host> parse_opaque_host(std::string_view input) {
+  bool parse_opaque_host(std::optional<std::string>& out, std::string_view input) {
```

Comment:
returning `std::optional` should in theory be just as fast as returning a boolean + taking the value type as a string. in practice, returned values (if they aren’t small primitives like `int`) are implemented by passing them as references to the function itself